### PR TITLE
Finish #59: Remove Visited Users from Compatibility List

### LINKED ISSUE #59: Remove visited users from compatibility list
Regenerate compatibility list should remove users that current user has already visited

### PR DESCRIPTION
**Pull request by nicholaspoon03**
# Description

Fixes #59 

* Added a method to remove visited users from the compatibility list that is called every time the compatibility list is generated.
* Added tests from removeVisitedUser and removeCurrentUser

# Checklist

Please do a self-check to make sure you have satisfied all of the following before marking your PR
as ready for review (place an `x` (the letter x) between `[ ]` to mark complete).

- [X] My PR does not contain unrelated files that may cause conflicts
- [X] I have performed a self-review of my own code
- [X] I have commented my code with javadoc and perhaps in-line comments (particularly in
  hard-to-understand areas)
- [X] My changes generate no new warnings or errors
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have not violated clean architecture principles
- [X] I have opted to use the most general form of datastructures (example: use `List` instead of
  `ArrayList` wherever possible)

### CODE REVIEW

**Review by rsrinivasan1 — APPROVED** [review]
Look great! Appreciate the clean list of methods in ```calculateCompatibilityList```, makes it easier to look at!

File: app/src/main/java/com/group80/uoftinder/entities/User.java [hunk 1]
```diff
@@ -213,4 +215,21 @@ public List<String> getMatches() {
...
         this.matches = matches;
     }
+
+    /**
+     * Indicates whether some other object has the same uid to this one
+     * @param obj: the object that is being compared to
+     * @return whether some other object has the same uid to this one
+     */
+    @Override
+    public boolean equals(@Nullable Object obj) {
```
Comment (helen-li) [inline app/src/main/java/com/group80/uoftinder/entities/User.java:225]: Nice application of what we learned about comparators and .equals!

**Review by helen-li — APPROVED** [review]
Looks good! Thanks for the quick turnaround regarding this bug Nick :)

**Review by chloe2407 — APPROVED** [review]
Thanks for making these changes Nicholas and also updating the visited and likes list into a list of strings instead of users!

**Pull request merged**